tensorflow/contrib/framework/python/framework/graph_util_test.py

```diff
@@ -57,5 +60,17 @@ def testGraphUtil(self):
     self.assertEqual(fused_graph_def.node[3].name, 'E')
 
 
+class GetPlaceholdersTest(test.TestCase):
+
+  def test_get_placeholders(self):
+    with ops.Graph().as_default() as g:
+      ids = []
+      for i in range(5):
+        x = array_ops.placeholder(dtypes.float32)
+        ids.append(x._id)
+      result = graph_util.get_placeholders(g)
+      result_ids = { i._id for i in result }
```

Comment:
why do we cast this to set? if result would return same placeholder several times, that would still pass this test while being incorrect behavior (I assume..). If ordering doesn't matter we can use sorted function

Reply:
Thank you for pointing out this.